model.py

```diff
@@ -25,66 +25,57 @@ class Genre(db.Model):
 
     __tablename__ = "genres"
 
-    genre_id = db.Column(db.Integer,
-                        autoincrement=True,
-                        primary_key=True)
+    genre_id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String(64), nullable=True)
 
     def __repr__(self):
         """Provide helpful representation when printed."""
 
         return f"<Genre genre_id={self.genre_id} name={self.name}>"
 
-# association_table = db.Table('association', 
-#     db.Column('location_id', db.Integer, db.ForeignKey('locations.location_id'), primary_key=True),
-#     db.Column('movie_id', db.Integer, db.ForeignKey('movies.movie_id'),primary_key=True),  
-# )
-#maybe explore this more later
+
+association_table = db.Table('movies_locations',
+                             db.Column('location_id', db.Integer, db.ForeignKey('locations.location_id')),
+                             db.Column('movie_id', db.Integer, db.ForeignKey('movies.movie_id')),
+                             )
+
 
 class Movie(db.Model):
     """Movie on ratings website."""
 
     __tablename__ = "movies"
 
-    movie_id = db.Column(db.Integer,
-                         autoincrement=True,
-                         primary_key=True)
+    movie_id = db.Column(db.Integer, primary_key=True)
     title = db.Column(db.String(100))
     genre_id = db.Column(db.Integer, db.ForeignKey('genres.genre_id'), nullable=False)
-
-    
     genre = db.relationship('Genre', backref='movies')
-    location_ids = db.Column(ARRAY(db.Integer))
```

Comment:
Not needed because we can use the `movies_locations` table with sqlalchemy to create this list for us. Notice I've removed `Movie.location_ids` and added `Movie.locations` instead.